web/vue-app/src/App.vue

```diff
@@ -1,28 +1,113 @@
 <template>
   <div id="app">
-    <img alt="Vue logo" src="./assets/logo.png">
-    <HelloWorld msg="Welcome to Your Vue.js App"/>
+    <div id="more-rent" class="results-gif" v-if="dataReceived && goodResult"></div>
+    <div id="less-rent" class="results-gif" v-if="dataReceived && (goodResult === false)"></div>
+    <results-component v-if="dataReceived" v-bind:inputs="inputs" />
+    <indexation-form   v-else @add:input="sendInputs" />
   </div>
 </template>
 
 <script>
-import HelloWorld from './components/HelloWorld.vue'
+import IndexationForm from './components/IndexationForm.vue'
+import ResultsComponent from './components/ResultsComponent.vue'
 
 export default {
   name: 'App',
   components: {
-    HelloWorld
+    IndexationForm,
+    ResultsComponent
+  },
+  data() {
+    return {
+      inputs: [
+        {
+          id: 1,
+          start: "01/09/2010",
+          signed: "25/07/2010",
+          region: "brussels",
+          rent: 500,
+          base_index: 1112.74,
+          current_index: 131.72,
+          new_rent: 584.18
+        },
+      ],
+      dataReceived: false,
+      goodResult: false
+    }
+  },
+  methods: {
+    addInput(input) {
+      const lastId =
+        this.inputs.length > 0
+          ? this.inputs[this.inputs.length - 1].id
+          : 0;
+      const id = lastId + 1;
+      const newInput = { ...input, id };
+
+      this.inputs = [...this.inputs, newInput];
+  },
+  async sendInputs(input) {
+    try {
+      const response = await fetch('http://localhost:4567/v1/indexations', {
+        method: 'POST',
+        body: JSON.stringify(input),
+        headers: {
+          'Content-type': 'application/json; charset=UTF-8' },
+      })
+      const data = await response.json()
+      // at this point, data is the response from sinatra
+      let newLine = {
+          rent: input.rent,
+          new_rent: data.new_rent,
+          base_index: data.base_index,
+          current_index: data.current_index
+      }
+      this.addInput(newLine)
+
+      this.inputs = [newLine]
```

Comment:
calling add input is not really useful as you are resetting `this.inputs = [newLine]` on the next line. 
```suggestion
      this.inputs = newLine
```
As you are always resetting `this.inputs` it might be a better option to just save the js object instead of an array

Reply:
Absolutely! Originally, I had a form, for which I was saving the inputs, hence why I was using an array. Later, when I came to the actual catching of the inputs and sending them to the back-end, I had no "immediate-use" for keeping the inputs since I had tested that I was indeed catching them correctly. I forgot to go back and trim more fat off the code from the earlier version :P Thank you for spotting!